pkg/provider/azure_storageaccount.go

```diff
@@ -94,6 +95,10 @@ func (az *Cloud) getStorageAccounts(accountOptions *AccountOptions) ([]accountWi
 				if !found {
 					continue
 				}
+
+				if accountOptions.EnableLargeFileShare && (len(acct.LargeFileSharesState) == 0 || acct.LargeFileSharesState == storage.LargeFileSharesStateDisabled) {
```

Comment:
the matching should happen out side of `if len(accountOptions.VirtualNetworkResourceIDs) > 0`, and premium account already supports LargeFileShare by default, so should skip matching for premium